### PR TITLE
lite-apiserver support cert rotation

### PR DESCRIPTION
**Pull request by Beckham007**
Based on #43 , lite-apiserver support cert rotation.

### CODE REVIEW

File: pkg/lite-apiserver/cert/cert_mgr.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+func (cm *CertManager) Start() {
+	t := time.NewTimer(cm.getReloadDuration())
+
+	go func() {
+		for {
+			fastReload := false
+
+			select {
+			case <-t.C:
+				for i := range cm.tlsConfig {
```
Comment (dixudx) [inline pkg/lite-apiserver/cert/cert_mgr.go:83]: Better to have these in a new function, which will be easier for unit tests as well.

File: pkg/lite-apiserver/transport/transport_mgr.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	}
+
+	return pool, nil
+}
+
+func makeTransport(tlsClientConfig *tls.Config) *http.Transport {
+	// TODO enable http2 if using go1.15
+	return &http.Transport{
+		DialContext: (&net.Dialer{
+			Timeout:   30 * time.Second,
```
Comment (dixudx) [inline pkg/lite-apiserver/transport/transport_mgr.go]: Make these setting as constant variables.

Or we could make them configurable with default values.

File: pkg/lite-apiserver/options/server_options.go [hunk 3]
```diff
@@ -50,11 +50,16 @@ func (s *RunServerOptions) ApplyTo(c *config.LiteServerConfig) error {
...
 	c.Port = s.Port
-	c.InsecurePort = s.InsecurePort
 	c.SyncDuration = time.Duration(s.SyncDuration) * time.Second
 	c.FileCachePath = s.FileCachePath
 	c.BackendTimeout = s.BackendTimeout
 
+	if len(s.ApiserverCAFile) > 0 {
+		c.ApiserverCAFile = s.ApiserverCAFile
+	} else {
+		c.ApiserverCAFile = s.CAFile
```
[the remark this answers is not in the document]
Comment (Beckham007) [reply pkg/lite-apiserver/options/server_options.go:60]: CAFile was checked in "RunServerOptions.Validate()"

File: pkg/lite-apiserver/options/server_options.go [hunk 4]
```diff
@@ -96,12 +101,12 @@ func (s *RunServerOptions) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.CAFile, "ca-file", "", "")
 	fs.StringVar(&s.CertFile, "tls-cert-file", "", "")
 	fs.StringVar(&s.KeyFile, "tls-private-key-file", "", "")
+	fs.StringVar(&s.ApiserverCAFile, "apiserver-ca-file", "", "")
```
[the remark this answers is not in the document]
Comment (Beckham007) [reply pkg/lite-apiserver/options/server_options.go]: done. thx

File: pkg/lite-apiserver/transport/transport_mgr.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+limitations under the License.
+*/
+
+package transport
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"github.com/superedge/superedge/pkg/lite-apiserver/cert"
```
[the remark this answers is not in the document]
Comment (Beckham007) [reply pkg/lite-apiserver/transport/transport_mgr.go]: done

File: pkg/lite-apiserver/cert/cert_mgr.go [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+			// load one cert error.
+			fastReload = true
+			continue
+		}
+
+		cm.updateCert(commonName, tlsCert)
+	}
+
+	cm.fastReload = fastReload
+	return nil
```
[the remark this answers is not in the document]
Comment (Beckham007) [reply pkg/lite-apiserver/cert/cert_mgr.go:72]: the error has logged in loadCert()

File: pkg/lite-apiserver/cert/cert_mgr.go (hunk 1 above)
Comment (Beckham007) [reply pkg/lite-apiserver/cert/cert_mgr.go:84]: add new function handleCertUpdate()

File: pkg/lite-apiserver/cert/cert_mgr.go [hunk 7]
```diff
@@ -66,7 +66,7 @@ func (cm *CertManager) Init() error {
 			continue
 		}
 
-		if  tlsCert.Leaf.NotAfter.After(time.Now()) {
+		if  tlsCert.Leaf.NotAfter.Before(time.Now()) {
```
[the remark this answers is not in the document]
Comment (dixudx) [reply pkg/lite-apiserver/cert/cert_mgr.go:69]: Or change to `time.Now().After(tlsCert.Leaf.NotAfter)` for more readable

[the remark this answers is not in the document]
Comment (dixudx) [reply pkg/lite-apiserver/cert/cert_mgr.go:69]: Also we should check whether the certificate is valid now.

`time.Now().Before(tlsCert.Leaf.NotBefore)`

[the remark this answers is not in the document]
Comment (Beckham007) [reply pkg/lite-apiserver/cert/cert_mgr.go]: done. thx for your suggestion.

**Review by dixudx — APPROVED** [review]
/lgtm

**Pull request merged**